pkg/controller/certificates/authority/policies.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authority
+
+import (
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	capi "k8s.io/api/certificates/v1beta1"
+)
+
+// SigningPolicy validates a CertificateRequest before it's signed by the
+// CertificateAuthority. It may default or otherwise mutate a certificate
+// template.
+type SigningPolicy interface {
+	// not-exporting apply forces signing policy implementations to be internal
+	// to this package.
+	apply(template *x509.Certificate) error
+}
+
+// PermissiveSigningPolicy is the signing policy historically used by the local
+// signer. It permits CertificateRequests with any SANs and applies a TTL.
```

Comment:
should we enforce things like "IsCA: false" and no extensions here, rather than in the signer setup of the CR template?

Reply:
Ya, I was thinking about it. We might want to reserve some rights in the CA, but I don't see a need for that now.

Reply:
document the extension and ca behavior of this signing policy

Reply:
Done.